File: Go/Solutions/projeuler/Problem036.go
package euler

import (
	"fmt"
	"strconv"
	"time"
)

/*
Problem036 answers the problem at : https://projecteuler.net/problem=36

* Problem 36:

The decimal number, 585 = 10010010012 (binary), is palindromic in both bases.

Find the sum of all numbers, less than one million, which are palindromic in base 10 and base 2.

**(Please note that the palindromic number, in either base, may not include leading zeros.)**

*/
func Problem036() {
	start := time.Now()
	ans := 0

	for i := 1; i < 1000000; i++ {
		k := strconv.Itoa(i)
		if isPallStr(k) && isPallStr(bin(int64(i))) {
			ans += i
		}
	}

	end := time.Now()
	fmt.Printf("\nAnswer to Problem 36 : %d\n", ans)
	fmt.Printf("Time Taken: %f seconds\n\n", end.Sub(start).Seconds())
}

File: Python/Problem036.py
from time import time

''' 
Problem 36: https://projecteuler.net/problem=36

The decimal number, 585 = 10010010012 (binary), is palindromic in both bases.

Find the sum of all numbers, less than one million, which are palindromic in base 10 and base 2.

(Please note that the palindromic number, in either base, may not include leading zeros.)
'''

pall = lambda n : str(n) == str(n)[::-1]
check = lambda n : pall(n) and pall(bin(n).replace('0b', ''))
start = time()
ans = sum([i for i in range(1,10**6) if check(i)])

if __name__=="__main__":
    print(f"\nAnswer: { ans }")
    print(f"Time Taken: { time() - start }\n")

